verifiers/internal/gha/verifier.go

```diff
@@ -80,11 +72,72 @@ func (v *GHAVerifier) VerifyArtifact(ctx context.Context,
 	return r, builderID, err
 }
 
+// VerifyArtifact verifies provenance for an artifact.
+func (v *GHAVerifier) VerifyArtifact(ctx context.Context,
+	provenance []byte, artifactHash string,
+	provenanceOpts *options.ProvenanceOpts,
+	builderOpts *options.BuilderOpts,
+) ([]byte, string, error) {
+	rClient, err := rekor.NewClient(defaultRekorAddr)
+	if err != nil {
+		return nil, "", err
+	}
+
+	/* Verify signature on the intoto attestation. */
+	env, cert, err := VerifyProvenanceSignature(ctx, rClient, provenance, artifactHash)
+	if err != nil {
+		return nil, "", err
+	}
+
+	return verifyEnvAndCert(env, cert, provenanceOpts, builderOpts)
+}
+
 // VerifyImage verifies provenance for an OCI image.
 func (v *GHAVerifier) VerifyImage(ctx context.Context,
```

Comment:
here it's called "image", in the CLI it's called "reference". Let's try to use the same terminology throughout the code.

Reply:
Going to move over to image. I'm not totally sure I can test verification via image and implicit tag (latest), image and explicit tag (reference), or digest locally, but i'll see if we can in e2e tests.